tools/src/main/scala/scala/scalanative/optimizer/pass/DeadCodeElimination.scala

```diff
@@ -32,9 +40,39 @@ class DeadCodeElimination(implicit top: Top) extends Pass {
 
       Seq(defn.copy(insts = newinsts))
   }
+
+  private def buildParamChanges(
```

Comment:
Computing which parameters are used and not is really a task of `analysis.UseDef`. We should move the code there and integrate it well into existing use def information.